tt_metal/impl/device/device.hpp

```diff
@@ -106,12 +106,17 @@ class Device {
 
     CoreCoord compute_with_storage_grid_size() const;
 
+    CoreCoord dram_grid_size() const;
+
     CoreCoord physical_core_from_logical_core(const CoreCoord &logical_core, const CoreType &core_type) const;
     CoreType core_type_from_physical_core(const CoreCoord &physical_core) const;
-    CoreCoord worker_core_from_logical_core(const CoreCoord &logical_core) const;
 
+    CoreCoord worker_core_from_logical_core(const CoreCoord &logical_core) const;
     std::vector<CoreCoord> worker_cores_from_logical_cores(const std::vector<CoreCoord> &logical_cores) const;
 
+    CoreCoord dram_core_from_logical_core(const CoreCoord &logical_core) const;
```

Comment:
do you think we need some way to map dram channel/bank to dram logical core?

Reply:
Might be useful for optimal placement? I can add it